frontend/src/components/profile-preferences-page/common/profile-preferences-page-form.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  FC,
+  ProfileUpdateRequestDto,
+  ProfileUpdateResponseDto,
+  UpdateProfileValue,
+  UserBaseResponseDto,
+} from 'common/types/types';
+import style from '../styles.module.scss';
+import React, { useCallback, useState } from 'react';
+import { profileActions } from 'store/actions';
+import { useAppDispatch, useAppForm } from 'hooks/hooks';
+import { profileUpdateValidationSchema } from '../../../validation-schemas/validation-schemas';
+import { Input } from '../../common/input/input';
+import { store } from '../../../store/store';
+import { errorMessages } from '../../../common/enums/messages';
+import { Loader } from '../../common/common';
+
+type Props = {
+  user: UserBaseResponseDto;
+  profile: ProfileUpdateResponseDto;
+};
+
+const ProfilePreferencesPageForm: FC<Props> = ({ user, profile }) => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const dispatch = useAppDispatch();
+  const { id: userId } = user;
+  const { firstName, lastName, username } = profile;
+  const [error, setError] = useState<string>();
+
+  const { control, errors, handleSubmit } = useAppForm<UpdateProfileValue>({
+    defaultValues: { firstName, lastName, username },
+    validationSchema: profileUpdateValidationSchema,
+  });
+
+  const handleUpdateProfileDataSubmit = useCallback(
+    async (payload: ProfileUpdateRequestDto) => {
+      try {
+        setIsLoading(true);
+        await dispatch(profileActions.updateProfile(payload)).unwrap();
+      } catch {
+        setError(store.getState().auth.error || errorMessages.DEFAULT);
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    [dispatch],
+  );
+
+  const onSubmit = async (submitValue: UpdateProfileValue): Promise<void> => {
+    const request: ProfileUpdateRequestDto = {
+      ...submitValue,
+      userId,
+    };
+    await handleUpdateProfileDataSubmit(request);
+  };
```

Comment:
Move all this logic to the parent component.

Reply:
done